File: src/dtos/productDto.ts

export class ProductDto {
  id?: string;
  name?: string;
  description?: string;
  image?: string;
  price?: number;

  constructor(prod){
    this.id = prod.id;
    this.name = prod.name;
    this.description = prod.description;
    this.image = prod.image;
    this.price = prod.price;
  }

  }

File: Api/ApiModels/Products/Response/ProductDto.cs
using System.ComponentModel.DataAnnotations;

public class ProductDto{
    public Guid Id { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public double Price { get; set; }
    public string? ImageUrl { get; set; }
    public string? Category { get; set; }

}